File: PCH/DP/BOJ_11726.py
import sys
input = sys.stdin.readline

n = int(input())

dp = [0]*1001
dp[1] = 1
dp[2] = 2
dp[3] = 3

if n >= 3:
  for i in range(3, n+1):
    dp[i] = dp[i-1] + dp[i-2]

print(dp[n]%10007)

File: LJH/DP/BOJ11726.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./LJH/input.txt";
let n = Number(fs.readFileSync(filePath).toString());
let dp = {
    1: 1,
    2: 2,
};
for (let i = 3; i <= n; i++) {
    dp[i] = (dp[i - 1] + dp[i - 2]) % 10007;
}
console.log(dp[n]);
